db/migrate/20190708192323_fix_unserializable_notification_options.rb

```diff
@@ -0,0 +1,44 @@
+class FixUnserializableNotificationOptions < ActiveRecord::Migration[5.1]
+  class Notification < ActiveRecord::Base
+    belongs_to :subject, :polymorphic => true
+    belongs_to :notification_type, :class_name => "FixUnserializableNotificationOptions::NotificationType"
+  end
+
+  class NotificationType < ActiveRecord::Base
+  end
+
+  class CloudVolume < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    Notification.where.not(:options => nil).includes(:subject).all.each do |n|
+      begin
+        # Options could have no longer available activerecord/activemodel classes from rails 5.0.0 -> 5.0.6, which will raise an ArgumentError on deserialization
+        # Fixing these to use 5.0.7+ smaller serialization format without all the private attributes is too hard, especailly since you can only read their YAML strings
+        # and make changes as a string, not as a Hash.
+        opts = YAML.load(n.options)
+      rescue ArgumentError
+        new_value = n.subject.try(:name) || n.subject.try(:description)
```

Comment:
Because subject is polymorphic, I believe I have to enumerate all classes that could be the subject of the notification as migration stubs to avoid autoloading the class from the filesystem.  With provider classes, this could be a large number of classes.  Do we care about that here?

Reply:
I can avoid the call to `.subject` entirely and just blank out the options here and 👇 on line 37...

Reply:
tldr;  I tried to retain anything we have in the notification to help identify the subject of the notification... this could cause a bloated migration with all possible polymorphic subject base classes or I can just give up and clear the options if it's unreadable.